src/Graphics/ColorBufferReader.cpp

```diff
@@ -18,17 +18,18 @@ namespace graphics {
 	const u8* ColorBufferReader::_convertFloatTextureBuffer(const u8* _gpuData, u32 _width, u32 _height,
 		u32 _heightOffset, u32 _stride)
 	{
-		std::copy_n(_gpuData, m_tempPixelData.size(), m_tempPixelData.data());
+		int bytesToCopy = m_pTexture->realWidth * _height * 16;
+		std::copy_n(_gpuData, bytesToCopy, m_tempPixelData.data());
 		u8* pixelDataAlloc = m_pixelData.data();
 		float* pixelData = reinterpret_cast<float*>(m_tempPixelData.data());
 		const u32 colorsPerPixel = 4;
 		const u32 widthPixels = _width * colorsPerPixel;
 		const u32 stridePixels = _stride * colorsPerPixel;
 
-		for (u32 index = 0; index < _height; ++index) {
+		for (u32 heightIndex = 0; heightIndex < _height && (heightIndex * widthPixels + widthPixels < m_pixelData.size()); ++heightIndex) {
```

Comment:
If  (heightIndex * widthPixels + widthPixels >= m_pixelData.size()) really possible, it is better to reduce _height instead of that ugly check:
```
if (_height * widthPixels  > m_pixelData.size())
  _height = m_pixelData.size() / widthPixels;
```